### ISSUE TITLE
Doors do not appear in the creative inventory.

### ISSUE THREAD

**Issue by C1ffisme** [minetest/minetest_game#1678]
![screenshot_20170330_203856](https://cloud.githubusercontent.com/assets/9893266/24532808/ec0c06e8-1588-11e7-89a0-952cbd5df4ee.png)

Interestingly enough, I looked at the code, and craftitems don't get the `not_in_creative_inventory` group applied to them.

**Comment by paramat** [comment]
Confirmed.

**Comment by paramat** [comment]
Possibly broken by https://github.com/minetest/minetest_game/commit/b847888cb7b79787f424dec2c2ee1728cc2c0872

**Comment by paramat** [comment]
Confirmed.
Broken by https://github.com/minetest/minetest_game/commit/b847888cb7b79787f424dec2c2ee1728cc2c0872
#1670

**Comment by C1ffisme** [comment]
I think the culprit might be (Line 363): https://github.com/minetest/minetest_game/commit/b847888cb7b79787f424dec2c2ee1728cc2c0872#diff-82ce76da0c47bb1b6754e4277adb052aR363

I might suggest using "itemgroups" for the craftitem, rather than using both for the door.

**Comment by paramat** [comment]
Line 363 is ok, doors appear in creative inventory with that line.

**Comment by SmallJoker** [comment]
Could someone please test this patch? It worked with unified_inventory here.
```Lua
 mods/doors/init.lua | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mods/doors/init.lua b/mods/doors/init.lua
index 8ca26da..3958444 100644
--- a/mods/doors/init.lua
+++ b/mods/doors/init.lua
@@ -254,7 +254,7 @@ function doors.register(name, def)
 	minetest.register_craftitem(":" .. name, {
 		description = def.description,
 		inventory_image = def.inventory_image,
-		groups = def.groups,
+		groups = table.copy(def.groups),
 
 		on_place = function(itemstack, placer, pointed_thing)
 			local pos
```

**Comment by paramat** [comment]
^ That does make doors re-appear in creative inventory.

**Comment by sofar** [comment]
The `table.copy()` is good, that's an OK fix.

**Comment by SmallJoker** [comment]
#1683 merged (b377aaa)
